dev/setup/shell.php

```diff
@@ -63,20 +63,15 @@ function get_terminal_lines_cols() {
 			$columns = $m['val'];
 		}
 	} else {
-		exec( 'resize', $output, $status );
+		$tput_exists = ! empty( exec( "command -v tput" ) );
```

Comment:
You are checking the output here, not the return status, which is what should be checked here, no matter the command. The output of a not found command will be a string like `tput: command not found` that would not be empty, thus affirming, incorrectly, that `$tput_exists`.

I feel this logic could be conflated by running less commands:

1. `exec( 'tput lines`), $lines, $status)` would both check if the command exists and get the lines.
2. if `$status !== 0` then bail, else `exec( 'tput cols', $cols, $status )`
3. if `$status === 0` again then process the lines and cols array outputs, else bail

Reply:
> The output of a not found command will be a string like `tput: command not found`

`command -v` returns which program in `$PATH` would execute the command. If none is found, it just returns an empty string.

> I would not remove the resize option as it's faster, when available, than the tput one (it's one command, not two).

3 milliseconds faster :shrug: 

I will commit the 1.2.3 suggestions soon

Reply:
The time diff is on Linux, it's sensible on Mac.

In your changes you dropped `resize` completely, I'd keep it as a fallback.